caas/clientconfig/k8s_test.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caas_test
+
+import (
+	"io/ioutil"
+	"os"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	caascfg "github.com/juju/juju/caas/clientconfig"
+	"github.com/juju/juju/cloud"
+	"github.com/juju/testing"
+)
+
+type k8sConfigSuite struct {
+	testing.IsolationSuite
+	reader caascfg.K8SClientConfigReader
+}
+
+var _ = gc.Suite(&k8sConfigSuite{})
+
+var (
+	emptyConfig = `
+apiVersion: v1
+kind: Config
+clusters: []
+contexts: []
+current-context: ""
+preferences: {}
+users: []
+`
+
+	singleConfig = `
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    server: https://1.1.1.1:8888
+  name: the-cluster
+contexts:
+- context:
+    cluster: the-cluster
+    user: the-user
+  name: the-context
+current-context: the-context
+preferences: {}
+users:
+- name: the-user
+  user:
+    password: thepassword
+    username: theuser
+`
+	multiConfig = `
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    server: https://1.1.1.1:8888
+  name: the-cluster
+- cluster:
+    server: https://10.10.10.10:1010
+  name: default-cluster
+contexts:
+- context:
+    cluster: the-cluster
+    user: the-user
+  name: the-context
+- context:
+    cluster: default-cluster
+    user: default-user
+  name: default-context
+current-context: default-context
+preferences: {}
+users:
+- name: the-user
+  user:
+    client-certificate-data: QQ==
+    client-key-data: Qg==
+- name: default-user
+  user:
+    password: defaultpassword
+    username: defaultuser
+- name: third-user
+  user:
+    token: "atoken"
+- name: fourth-user
+  user:
+    client-certificate-data: QQ==
+    client-key-data: Qg==
+    token: "tokenwithcerttoken"
+- name: fifth-user
+  user:
+    client-certificate-data: QQ==
+    client-key-data: Qg==
+    username: "fifth-user"
+    password: "userpasscertpass"
+ 
+`
+)
+
+// writeTempKubeConfig writes yaml to a temp file and sets the
+// KUBECONFIG environment variable so that the clientconfig code reads
+// it instead of the default.
```

Comment:
I wonder if it might be better if `GetClientConfig` took a path to read from and a separate helper existed to locate the config file. That would make testing a whole lot easier.